lib/OpenQA/Schema/ResultSet/Assets.pm

```diff
@@ -28,9 +28,17 @@ sub register ($self, $type, $name, $options = {}) {
         log_warning "no file found for asset '$name' type '$type'";
         return undef;
     }
-    $self->result_source->schema->txn_do(
+    my $schema = $self->result_source->schema;
+    my $sth = $schema->storage->dbh->prepare_cached(
+        <<'END_SQL'
```

Comment:
I think you can still do `my $sth = $schema->storage->dbh->prepare_cached(<<'END_SQL');` at least.